src/transformers/generation_tf_utils.py

```diff
@@ -184,10 +197,10 @@ def generate(
         """
 
         # We cannot generate if the model does not have a LM head
-        if self.get_output_embeddings() is None:
+        is_lm, list_models = self.is_lm_model()
```

Comment:
Way cleaner, thanks!

Reply:
I don't really agree here -> I prefer to leave the `self.get_output_embeddings()` check here. E.g. `TFRag` won't be in any of the `TF_MODEL_FOR_CAUSAL_LM_MAPPING, TF_MODEL_FOR_MASKED_LM_MAPPING, TF_MODEL_FOR_SEQ_TO_SEQ_CAUSAL_LM_MAPPING`. And since `TFRag` will have two models of that can both generate, but won't both be able to be in the same MAPPING class we will already run into problems here. Also this is inconsistent with PyTorch.

Logically, I like this check as it is right now. If the model has `output_embeddings` then it can generate because all one needs for generate is a logit output vector. I don't want to tie this functionality to the LM Mappings in the `modeling_auto.py`. It creates an unnecessary dependency IMO.

Reply:
In order to be sure, `self.get_output_embeddings()` is nowhere used except for a simple check in the `generation_tf_utils.py`, so what I deduce from that is that it is a kind of useless method. Unless they are used somewhere else?

For sure we can create a smarter way to detect if a model has an LM layer, if the role of `self.get_output_embeddings()` is just to know that. What could be the best compromise?